### PR TITLE
MODINREACH-260 - Final Check In Fails at Borrowing Site if Transaction State is ITEM_RECEIVED

### PR DESCRIPTION
**Pull request by s1mpleGit**
[MODINREACH-260](https://issues.folio.org/browse/MODINREACH-260) - Final Check In Fails at Borrowing Site if Transaction State is ITEM_RECEIVED

## Purpose
The "finalcheckin" message on an INN-Reach transaction should ALWAYS be accepted by the borrowing site, regardless of transaction status (except PATRON_HOLD and TRANSFER, as those indicate that the item has not yet left the owning site, and could just be cancelled to terminate the transaction). This allows for otherwise stuck transactions, or transactions where proper receiving and check-out processes have not been followed, to be closed out when the associated item is returned to the owning site through whatever process that might occur. Any open FOLIO requests or loans at the borrowing site should be closed - cancelled. FOLIO requests closed with INN-Reach cancellation reason.

## Approach
"finalcheckin" message is sent to the central by the owning site and received by the borrowing site.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java [hunk 1]
```diff
@@ -322,11 +322,11 @@ private void updateTransactionOnLoanClosure(StorageLoanDTO loan, InnReachTransac
 
       hold.setDueDateTime(null);
 
-      clearCentralPatronInfo(transaction.getHold());
-
       transaction.setState(FINAL_CHECKIN);
 
       notifier.reportFinalCheckIn(transaction);
+
+      clearCentralPatronInfo(hold);
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java]: Can you please explain how this change is going to fix the issue with handling "finalcheckin" message on borrowing site?

Comment (s1mpleGit) [reply src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java]: removed

File: src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java [hunk 2]
```diff
@@ -1052,13 +1052,18 @@ private void assertPatronHoldFieldsAreNull(TransactionPatronHold hold) {
...
 
-  @Test
+  @ParameterizedTest
+  @EnumSource(names = {"PATRON_HOLD", "TRANSFER"})
   @Sql(scripts = {
     "classpath:db/central-server/pre-populate-central-server.sql",
     "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql"
   })
-  void checkTransactionIsNotInStateItemInTransitOrReturnUncirculated() {
+  void checkTransactionIsInStatePatronHoldOrTransfer(InnReachTransaction.TransactionState testEnums) {
```
Comment (oliinyko) [inline src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java]: what is testEnums? lets rename it to state

Comment (s1mpleGit) [reply src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java]: renamed

**Review by oliinyko — APPROVED** [review]

**Pull request merged**